Library/OcCryptoLib/Sha2.c

```diff
@@ -494,6 +450,11 @@ Sha512 (
 {
   SHA512_CONTEXT  Ctx;
 
+  if (IsAvxSupported ()) {
```

Comment:
There are places where we use call Sha384Update directly, for example, where we update the hash from non-contiguous data. For this use the branching for AVX and non-AVX version needs to happen inside Sha384Update rather than here.